Project1/PartyBase.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+#include "Unit.h"
+#include <list>
+
+class Map;
+class Party;
+class party_iterator;
+using UnitRef = std::reference_wrapper<Unit>;
+using constUnitRef = std::reference_wrapper<const Unit>;
+struct UnitData;
+struct PartyData {
+	std::string name;
+	std::vector<UnitData> unit_data;
+};
+class PartyBase {
+protected:
+	using iterator = std::list<Unit>::iterator;
+	using const_iterator = std::list<Unit>::const_iterator;
+
+	std::string _party_name;
+	std::list<Unit> _units;
+
+	PartyBase();
+	PartyBase(std::string name);
+	PartyBase(std::string name, std::vector<UnitData> unit_data);
+	PartyBase(PartyData data);
+
+	const_iterator getPosition(const Unit& unit) const;
+	iterator getPosition(const Unit& unit);
+
+	void insertUnit(UnitData unit);
+	void insertUnit(Unit& unit);
```

Comment:
This should take `unique_ptr<Unit>`, because we want the Party to own it's units instead of just knowing about units someone else owns.